### PR TITLE
Add a function to get the full-size of xg and yg from input files

### PR DESCRIPTION
**Pull request by cspencerjones**
This function gets XG and YG from .mitgrid files. They are the full size (i.e. they include the points on the right hand side of the final gridpoint in both x and y directions. This is useful if you want to determine which tile and individual lat/lon point is on.

### CODE REVIEW

**Comment by raphaeldussin** [comment]
what was wrong with get_grid_from_input ?

https://github.com/MITgcm/xmitgcm/blob/1fb9cc2520d0df1cf199b347d94c91853d22dddb/xmitgcm/utils.py#L1247

**Comment by cspencerjones** [comment]
It doesn't do what I want - it doesn't get the rightmost gridpoints.

File: xmitgcm/test/test_utils.py [hunk 1]
```diff
@@ -973,6 +973,49 @@ def test_get_grid_from_input(all_grid_datadirs, usedask):
...
+    ds = get_xg_yg_from_input(dirname + '/' + expected['gridfile'],
+                             geometry=expected['geometry'],
+                             dtype=np.dtype('d'), endian='>',
+                             use_dask=usedask,
+                             extra_metadata=md,
+                             tilex=tx,tiley=ty,
+                             blankList=bl)
+    # test types
+    assert type(ds) == xarray.Dataset
+    assert type(ds['XG']) == xarray.core.dataarray.DataArray
```
Comment (rabernat) [inline xmitgcm/test/test_utils.py]: use `isinstance` here

**Review by rabernat — CHANGES_REQUESTED** [review]
Thanks for looking into this @cspencerjones. I appreciate that the existing `get_grid_from_inputs` function did not work for the reasons you stated. However, I would strongly prefer that we modify that function to meet your needs, rather than introduce a new one with much of the same code.

I see two separate features here:
- The ability to get the extra "outer" grid point from the mitgrid files. This could be an optional flag to `get_grid_from_inputs`, maybe `outer=False` as a default, to keep backwards compatibility
- The ability to reshape a dataset in tiles of a specified shape.

Could we do these as two separate functions, or is there some fundamental reason why they need to be in a single function?

**Comment by rabernat** [comment]
@cspencerjones - are you planning to continue to work on this?

**Comment by cspencerjones** [comment]
It's not my priority. If we think that these changes would be useful for other people then I can but I'm not sure that there is actually need for them right now. So probably no, I will just keep using my own branch. Do you have a strong opinion?

**Comment by AaronDavidSchneider** [comment]
Hi @cspencerjones,
I could really use the outer grid points (xESMF needs outer grid points for conservative regridding). I would be happy if this PR could be somehow resumed :D I'd be happy to help.

**Comment by cspencerjones** [comment]
@AaronDavidSchneider I do not have time to work on this at the moment, since my project has moved away from this problem, at least temporarily. The code is quite dense and hard to follow, and as @rabernat mentions above my new code actually repeats a lot of what's in the original function. However you are welcome to take it and see how it works for you. If it does, then maybe you could contribute to this pull request as well?

**Comment by rabernat** [comment]
Yes it would be great if @AaronDavidSchneider would be able to pick up this PR! We will be happy to help you.

From a technical standpoint, it is probably easiest for you to check out @cspencerjones's branch and make a new PR from your own fork.

**Comment by AaronDavidSchneider** [comment]
> From a technical standpoint, it is probably easiest for you to check out @cspencerjones's branch and make a new PR from your own fork.

That sounds good. I will try my best to get it working. I am not quite sure when I will have the time for it.
